code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -42,36 +45,73 @@
 			break
 
 		if(isliving(A))
-			var/mob/living/L = A
-			if((L.faction == src.faction) && !attack_same)
-				continue
-			else if(L in friends)
-				continue
-			else
-				if(!L.stat)
-					stance = HOSTILE_STANCE_ATTACK
-					T = L
-					break
+			T = find_weakest()
+			break
 
 		else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().
 			var/obj/mecha/M = A
 			if (M.occupant)
-				stance = HOSTILE_STANCE_ATTACK
 				T = M
 				break
 
 		if(istype(A, /obj/machinery/bot))
 			var/obj/machinery/bot/B = A
 			if (B.health > 0)
-				stance = HOSTILE_STANCE_ATTACK
 				T = B
 				break
 
 	if (T != target_mob)
 		target_mob = T
 		FoundTarget()
+	if(T != null)
+		stance = HOSTILE_STANCE_ATTACK
 	return T
 
+// Finds weakest(with least health) living mob to attack
+/mob/living/simple_animal/hostile/proc/find_weakest()
+	var/lowest_health = 100
+	var/atom/T = null
+	for(var/atom/A in targets)

```

Comment:
Since we're already doing `if (isliving(A))`. Change this loop to `for (var/mob/living/L in targets)`. It amounts to a more efficient operation.